plugins/module_utils/CertACLHelper.cs

```diff
@@ -0,0 +1,282 @@
+                        "Security Descr",
+                        new SafeSecurityDescriptorPtr(),
+                        0,
+                        ref securityDescriptorSize,
+                        SecurityInformationFlags.DACL_SECURITY_INFORMATION) != 0)
+                    {
+                        throw new Exception("Could not get security descriptor");
+                    }
+                    securityDescriptorBuffer = new SafeSecurityDescriptorPtr(securityDescriptorSize);
+
+                    if (NCryptGetProperty(
+                        handle,
+                        "Security Descr",
+                        securityDescriptorBuffer,
+                        securityDescriptorSize,
+                        ref securityDescriptorSize,
+                        SecurityInformationFlags.DACL_SECURITY_INFORMATION) != 0)
+                    {
+                        throw new Exception("Could not get security descriptor");
+                    }
+
+                }
+                else
+                {
+                    if (!CryptGetProvParam(
+                            handle,
+                            CryptProvParam.PP_KEYSET_SEC_DESCR,
+                            new SafeSecurityDescriptorPtr(),
+                            ref securityDescriptorSize,
+                            SecurityInformationFlags.DACL_SECURITY_INFORMATION))
+                    {
+                        throw new Win32Exception(Marshal.GetLastWin32Error());
+                    }
+
+
+                    securityDescriptorBuffer = new SafeSecurityDescriptorPtr(securityDescriptorSize);
+                    if (!CryptGetProvParam(
+                            handle,
+                            CryptProvParam.PP_KEYSET_SEC_DESCR,
+                            securityDescriptorBuffer,
+                            ref securityDescriptorSize,
+                            SecurityInformationFlags.DACL_SECURITY_INFORMATION))
+                    {
+                        throw new Win32Exception(Marshal.GetLastWin32Error());
+                    }
+
+                }
+                byte[] buffer = new byte[securityDescriptorSize];
+                Marshal.Copy(securityDescriptorBuffer.DangerousGetHandle(), buffer, 0, buffer.Length);
+                FileSecurity acl = new FileSecurity();
+                acl.SetSecurityDescriptorBinaryForm(buffer);
+
+                return acl;
+
+            }
+            set
+            {
+                if (ncrypt)
+                {
+                    byte[] sd = value.GetSecurityDescriptorBinaryForm();
+                    if (NCryptSetProperty(
+                        handle,
+                        "Security Descr",
+                        sd,
+                        (uint)sd.Length,
+                        SecurityInformationFlags.DACL_SECURITY_INFORMATION) != 0)
+                    {
+                        throw new Exception("Could not set security descriptor");
```

Comment:
We are throwing away the error code here, it seems like `NCryptSetProperty` returns the error code which we can pass into Win32Exception.